lib/Dialects/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -145,6 +145,27 @@ Operation *ReverseOp::getTiledImplementation(OpBuilder &builder,
 // TileOp
 //===----------------------------------------------------------------------===//
 
+void TileOp::build(mlir::OpBuilder &builder, mlir::OperationState &result,
+                   Value tileSize, ValueRange outSlices,
+                   TileOp::TileOpBodyBuilderFn bodyBuilder) {
+  result.addOperands(tileSize);
+  result.addOperands(outSlices);
+  result.addTypes(outSlices.getType());
+
+  Region *bodyRegion = result.addRegion();
+  bodyRegion->push_back(new Block);
+  Block &bodyBlock = bodyRegion->front();
+  bodyBlock.addArgument(builder.getIndexType());
+  bodyBlock.addArgument(builder.getIndexType());
+  bodyBlock.addArguments(outSlices.getType());
```

Comment:
The out arguments to the body don't have the same types as the outer arguments. E.g. if someone passes in `tensor<100xf32>` as an output of a `tile` op with 4 tiles, the out argument should have a type of either `tensor<25xf32>` or `tensor<?xf32>`.

Reply:
I was a bit overzealous with the "semantically meaningful naming" .. the slices are only the BBArgs.
Adding a test to confirm.